src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -154,9 +145,76 @@ internal WebApplicationBuilder(WebApplicationOptions options, bool slim, Action<
                 options.SuppressEnvironmentConfiguration = true;
             });
 
+        _genericWebHostServiceDescriptor = InitializeHosting(bootstrapHostBuilder);
+    }
+
+    internal WebApplicationBuilder(WebApplicationOptions options, bool slim, bool empty, Action<IHostBuilder>? configureDefaults = null)
```

Comment:
Nit: We only have two remaining readonly fields. Is it time to make those non readonly (and possibly "lie" about the nullability), give these more logical method names and remove the weird bool args. E.g. `InitializeEmpty`, `InitializeSlim`, `InitializeFull`?

Reply:
Let's consider this in a future change. I'm not excited about the factoring now, but it can easily be changed later without breaking anyone.